src/Simplify.cpp

```diff
@@ -4274,6 +4349,16 @@ void check_algebra() {
     check((x - y)*-2, (y - x)*2);
     check((xf - yf)*-2.0f, (yf - xf)*2.0f);
 
+    // Pull terms that are a multiple of the divisor out of a ternary expression
+    check(((x*4 + y) + z) / 2, x*2 + (y + z)/2);
+    check(((x*4 - y) + z) / 2, x*2 - (y - z)/2);
```

Comment:
I see your point. Then the previous rules that simplify divisions of binary expressions would be wrong in the same way. For example, `check((x*4 - y)/2, x*2 - y/2);`. Can we fix it by rearranging the parenthesis? Something like:
```
check((x*4 - y)/2, x*2  + (- y)/2);
check(((x*4 - y) + z) / 2, x*2 + (- y + z)/2);
```